code/modules/power/apc/apc_malf.dm

```diff
@@ -36,6 +36,9 @@
 		return
 	if(!is_station_level(z))
 		return
+	var/confirm = tgui_alert(malf, "Are you sure that you want to shunt? This will take you out of your core!", "Shunt to [name]?", list("Yes", "No"))
+	if(confirm != "Yes")
+		return
```

Comment:
malfoccupy isn't allowed to sleep. The way to fix this is to put this tgui_alert and everything below that into a new proc, then use `INVOKE_ASYNC()` to call that proc. code should have a few examples.

Reply:
makes sense